gdal/frmts/msg/msgdataset.cpp

```diff
@@ -345,31 +343,22 @@ MSGRasterBand::MSGRasterBand( MSGDataset *poDSIn, int nBandIn )
     // Find if we're dealing with MSG1, MSG2, MSG3 or MSG4
     // Doing this per band is the only way to guarantee time-series when the satellite is changed
 
-    std::string sPrologueFileName = poDSIn->command.sPrologueFileName(poDSIn->iCurrentSatellite, nBand);
+    int iCurrentSatellite = poDSIn->iCurrentSatellite; // Start with the satellite selected by the dataset
+
+    std::string sPrologueFileName = poDSIn->command.sPrologueFileName(iCurrentSatellite, 1);
     bool fPrologueExists = (access(sPrologueFileName.c_str(), 0) == 0);
 
-    // Make sure we're testing for MSG1,2,3 or 4 exactly once, start with the most recently used, and remember it in the static member for the next round.
-    if (!fPrologueExists)
+    // Make sure we're testing for MSG1,2,3 or 4 exactly once, start with the one of the dataset
+    int iTries = 1;
+    while (!fPrologueExists && (iTries < MAX_SATELLITES))
```

Comment:
Isn't this function exactly the same as in the dataset case?
We could extract it and call in both times.